src/main/java/org/assertj/core/internal/RealNumbers.java

```diff
@@ -58,4 +61,100 @@ protected boolean isGreaterThan(NUMBER value, NUMBER other) {
     return value.compareTo(other) > 0;
   }
 
-}
\ No newline at end of file
+  /**
+   * Checks whether specified value is a finite floating-point value.
+   * @param value the value to check.
+   * @return {@code true} if the value is a finite floating-point value; {@code false} otherwise.
+   */
+  protected abstract boolean isFinite(NUMBER value);
+
+  /**
+   * Verifies that the actual value is a finite floating-point value.
+   * @param info contains information about the assertion.
+   * @param actual the actual value.
+   * @throws AssertionError if the actual value is not a finite floating-point value.
+   */
+  public void assertIsFinite(AssertionInfo info, NUMBER actual) {
+    assertNotNull(info, actual);
+    final boolean result = isFinite(actual);
+    if (result) {
+      return;
+    }
+    throw failures.failure(info, shouldBeTrue(false));
```

Comment:
Add an error message specific to the assertion, for that create a class named `ShouldBeFinite`, take example of `ShouldBeOdd` and `ShouldBeOdd_create_Test`.